src/Extend/NotificationChannel.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Extension\Extension;
+use Flarum\User\NotificationPreference;
+use Illuminate\Contracts\Container\Container;
+
+class NotificationChannel implements ExtenderInterface
+{
+    /**
+     * @var string
+     */
+    private $channel;
+
+    private $enabled = [];
+
+    public function __construct(string $channel)
+    {
+        $this->channel = $channel;
+    }
+
+    public function extend(Container $container, Extension $extension = null)
```

Comment:
Super tiny thing, but I think it might be better to have the extend method after the enabled method. That way, code relating to the API isn't split up by the implementation.